src/NeoSharp.Core/Messaging/Handlers/AddrMessageMessageHandler.cs

```diff
@@ -29,8 +29,8 @@ public Task Handle(AddrMessage message, IPeer sender)
                 .ToArray();
 
             var endPointsToConnect = message.Payload.Address
-                .Select(nat => nat.EndPoint)
-                .Where(ep => connectedEndPoints.Contains(ep) == false) 
+                .Select(nat => new EndPoint(Protocol.Tcp, nat.EndPoint))
```

Comment:
Why cannot we send EndPoint directly?

Reply:
Because neo protocol only understand IpEndPoint